torch/distributed/pipelining/schedules.py

```diff
@@ -1186,6 +1180,12 @@ def step(self, *args, target=None, losses: Optional[List] = None, **kwargs):
         target: target for the loss function.
         losses: a list to store the losses for each microbatch.
         """
+        if not self._stages_initialized:
```

Comment:
hm, should I move this logic inside _step_microbatches? it seems some of our tests are calling _step_microbatches.  Do we allow this or do we require calling step()?

if we don't require calling step(), then should we also move this code inside _step_microbatches? 

```        # Clean per iteration
        self._stage.clear_runtime_states()
```

Reply:
Yeah. That would be a good place.
We may have to ask CUDA Graph to only come capture after the first step(...) though.
But that may not be a big deal? And the current change moved the prepare from init to step anyway.

Reply:
annoying: Stage class doesn't know "n_microbatches", which may have been intentional? 

I am thinking about whether it is bad to let the schedule 'register' the number of microbatches during Schedule.__init__ and then stage can use this value later when it performs initialization inside forward_one_chunk / backward_one_chunk?